To: vim-dev@vim.org
Subject: Patch 7.3.016
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.016
Problem:    Netbeans doesn't work under Athena.
Solution:   Support Athena, just like Motif. (Xavier de Gaye)
Files:      runtime/doc/netbeans.txt, src/gui.c, src/main.c, src/netbeans.c


*** ../vim-7.3.015/runtime/doc/netbeans.txt	2010-08-15 21:57:13.000000000 +0200
--- runtime/doc/netbeans.txt	2010-09-29 17:13:43.000000000 +0200
***************
*** 1,4 ****
! *netbeans.txt*  For Vim version 7.3.  Last change: 2010 Jul 20
  
  
  		  VIM REFERENCE MANUAL    by Gordon Prieur et al.
--- 1,4 ----
! *netbeans.txt*  For Vim version 7.3.  Last change: 2010 Aug 20
  
  
  		  VIM REFERENCE MANUAL    by Gordon Prieur et al.
***************
*** 118,124 ****
  uncommenting a line with "--disable-netbeans" in the Makefile.
  
  Currently the NetBeans interface is supported by Vim running in a terminal and
! by GVim when it is run with one of the following GUIs: GTK, GNOME, and Motif.
  
  If Motif support is required the user must supply XPM libraries.  See
  |workshop-xpm| for details on obtaining the latest version of XPM.
--- 118,125 ----
  uncommenting a line with "--disable-netbeans" in the Makefile.
  
  Currently the NetBeans interface is supported by Vim running in a terminal and
! by GVim when it is run with one of the following GUIs: GTK, GNOME, Windows,
! Athena and Motif.
  
  If Motif support is required the user must supply XPM libraries.  See
  |workshop-xpm| for details on obtaining the latest version of XPM.
*** ../vim-7.3.015/src/gui.c	2010-08-15 21:57:28.000000000 +0200
--- src/gui.c	2010-09-29 17:13:43.000000000 +0200
***************
*** 2352,2358 ****
      if (draw_sign)
  	/* Draw the sign on top of the spaces. */
  	gui_mch_drawsign(gui.row, col, gui.highlight_mask);
! # if defined(FEAT_NETBEANS_INTG) && (defined(FEAT_GUI_MOTIF) \
  	|| defined(FEAT_GUI_GTK) || defined(FEAT_GUI_W32))
      if (multi_sign)
  	netbeans_draw_multisign_indicator(gui.row);
--- 2352,2358 ----
      if (draw_sign)
  	/* Draw the sign on top of the spaces. */
  	gui_mch_drawsign(gui.row, col, gui.highlight_mask);
! # if defined(FEAT_NETBEANS_INTG) && (defined(FEAT_GUI_X11) \
  	|| defined(FEAT_GUI_GTK) || defined(FEAT_GUI_W32))
      if (multi_sign)
  	netbeans_draw_multisign_indicator(gui.row);
*** ../vim-7.3.015/src/main.c	2010-08-15 21:57:28.000000000 +0200
--- src/main.c	2010-09-29 17:13:43.000000000 +0200
***************
*** 940,946 ****
      if (netbeansArg != NULL && strncmp("-nb", netbeansArg, 3) == 0)
      {
  # ifdef FEAT_GUI
! #  if !defined(FEAT_GUI_MOTIF) && !defined(FEAT_GUI_GTK)  \
  		&& !defined(FEAT_GUI_W32)
  	if (gui.in_use)
  	{
--- 940,946 ----
      if (netbeansArg != NULL && strncmp("-nb", netbeansArg, 3) == 0)
      {
  # ifdef FEAT_GUI
! #  if !defined(FEAT_GUI_X11) && !defined(FEAT_GUI_GTK)  \
  		&& !defined(FEAT_GUI_W32)
  	if (gui.in_use)
  	{
*** ../vim-7.3.015/src/netbeans.c	2010-08-15 21:57:26.000000000 +0200
--- src/netbeans.c	2010-09-29 17:13:43.000000000 +0200
***************
*** 87,93 ****
  static void nb_init_graphics __ARGS((void));
  static void coloncmd __ARGS((char *cmd, ...));
  static void nb_set_curbuf __ARGS((buf_T *buf));
! #ifdef FEAT_GUI_MOTIF
  static void messageFromNetbeans __ARGS((XtPointer, int *, XtInputId *));
  #endif
  #ifdef FEAT_GUI_GTK
--- 87,93 ----
  static void nb_init_graphics __ARGS((void));
  static void coloncmd __ARGS((char *cmd, ...));
  static void nb_set_curbuf __ARGS((buf_T *buf));
! #ifdef FEAT_GUI_X11
  static void messageFromNetbeans __ARGS((XtPointer, int *, XtInputId *));
  #endif
  #ifdef FEAT_GUI_GTK
***************
*** 112,118 ****
  static NBSOCK nbsock = -1;		/* socket fd for Netbeans connection */
  #define NETBEANS_OPEN (nbsock != -1)
  
! #ifdef FEAT_GUI_MOTIF
  static XtInputId inputHandler = (XtInputId)NULL;  /* Cookie for input */
  #endif
  #ifdef FEAT_GUI_GTK
--- 112,118 ----
  static NBSOCK nbsock = -1;		/* socket fd for Netbeans connection */
  #define NETBEANS_OPEN (nbsock != -1)
  
! #ifdef FEAT_GUI_X11
  static XtInputId inputHandler = (XtInputId)NULL;  /* Cookie for input */
  #endif
  #ifdef FEAT_GUI_GTK
***************
*** 143,149 ****
  
      netbeans_send_disconnect();
  
! #ifdef FEAT_GUI_MOTIF
      if (inputHandler != (XtInputId)NULL)
      {
  	XtRemoveInput(inputHandler);
--- 143,149 ----
  
      netbeans_send_disconnect();
  
! #ifdef FEAT_GUI_X11
      if (inputHandler != (XtInputId)NULL)
      {
  	XtRemoveInput(inputHandler);
***************
*** 700,706 ****
  /*
   * Read a command from netbeans.
   */
! #ifdef FEAT_GUI_MOTIF
      static void
  messageFromNetbeans(XtPointer clientData UNUSED,
  		    int *unused1 UNUSED,
--- 700,706 ----
  /*
   * Read a command from netbeans.
   */
! #ifdef FEAT_GUI_X11
      static void
  messageFromNetbeans(XtPointer clientData UNUSED,
  		    int *unused1 UNUSED,
***************
*** 2937,2943 ****
      if (!NB_HAS_GUI || !NETBEANS_OPEN)
  	return;
  
! # ifdef FEAT_GUI_MOTIF
      /* tell notifier we are interested in being called
       * when there is input on the editor connection socket
       */
--- 2937,2943 ----
      if (!NB_HAS_GUI || !NETBEANS_OPEN)
  	return;
  
! # ifdef FEAT_GUI_X11
      /* tell notifier we are interested in being called
       * when there is input on the editor connection socket
       */
***************
*** 3027,3033 ****
      }
  }
  
! #if defined(FEAT_GUI_MOTIF) || defined(FEAT_GUI_W32) || defined(PROTO)
  /*
   * Tell netbeans that the window was moved or resized.
   */
--- 3027,3033 ----
      }
  }
  
! #if defined(FEAT_GUI_X11) || defined(FEAT_GUI_W32) || defined(PROTO)
  /*
   * Tell netbeans that the window was moved or resized.
   */
***************
*** 3458,3464 ****
      return FALSE;
  }
  
! #if defined(FEAT_GUI_MOTIF) || defined(PROTO)
  /*
   * We have multiple signs to draw at the same location. Draw the
   * multi-sign indicator instead. This is the Motif version.
--- 3458,3464 ----
      return FALSE;
  }
  
! #if defined(FEAT_GUI_X11) || defined(PROTO)
  /*
   * We have multiple signs to draw at the same location. Draw the
   * multi-sign indicator instead. This is the Motif version.
***************
*** 3487,3493 ****
      XDrawPoint(gui.dpy, gui.wid, gui.text_gc, x+3, y++);
      XDrawPoint(gui.dpy, gui.wid, gui.text_gc, x+2, y);
  }
! #endif /* FEAT_GUI_MOTIF */
  
  #if defined(FEAT_GUI_GTK) && !defined(PROTO)
  /*
--- 3487,3493 ----
      XDrawPoint(gui.dpy, gui.wid, gui.text_gc, x+3, y++);
      XDrawPoint(gui.dpy, gui.wid, gui.text_gc, x+2, y);
  }
! #endif /* FEAT_GUI_X11 */
  
  #if defined(FEAT_GUI_GTK) && !defined(PROTO)
  /*
*** ../vim-7.3.015/src/version.c	2010-09-29 16:55:45.000000000 +0200
--- src/version.c	2010-09-29 17:14:47.000000000 +0200
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     16,
  /**/

-- 
hundred-and-one symptoms of being an internet addict:
229. You spend so much time thinking what to add on this list.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\        download, build and distribute -- http://www.A-A-P.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
